FormalConjectures/ErdosProblems/107.lean

```diff
@@ -0,0 +1,121 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Erdős Problem 107
+
+*Reference:* [erdosproblems.com/107](https://www.erdosproblems.com/107)
+-/
+
+open Filter
+
+namespace Erdos107
+
+abbrev Point := EuclideanSpace ℝ (Fin 2)
+
+namespace Point
+
+/-- The three points `a,b,c` are not collinear. -/
+def InGenPos₃ (a b c : Point) : Prop :=
+  Matrix.det !![a 0, b 0, c 0 ; a 1, b 1, c 1 ; 1, 1, 1] ≠ 0
+
+/-- No three points in `S` are collinear. -/
+def SetInGenPos (S : Set Point) : Prop :=
+  ∀ ⦃p q r : Point⦄, {p,q,r} ⊆ S → List.Nodup [p,q,r] → InGenPos₃ p q r
```

Comment:
```suggestion
  ∀ ⦃p q r : Point⦄, {p,q,r} ⊆ S → {p,q,r}.ncard = 3 → InGenPos₃ p q r
```
Would that be closer to the informal "three"?
But in general I would use 
```
def NonTrilinear (A : Set ℝ²) : Prop := ∀ᵉ (x ∈ A) (y ∈ A) (z ∈ A),
  A.Triplewise (fun x y z ↦ ¬ Collinear ℝ {x, y, z})
```
and move that from https://github.com/google-deepmind/formal-conjectures/blob/ed41ca10bdce5343dfe037abc84befc685c86016/FormalConjectures/ErdosProblems/846.lean#L37 to ForMathlib. Then we can re-use it (also in FormalConjectures/ErdosProblems/213.lean)